controllers/protectedapplicationsview_controller.go

```diff
@@ -0,0 +1,567 @@
+			subRefs = append(subRefs, corev1.ObjectReference{
+				APIVersion: subscriptionAPIVersion,
+				Kind:       subscriptionKind,
+				Name:       sub.Name,
+				Namespace:  sub.Namespace,
+			})
+		}
+
+		var appRef corev1.ObjectReference
+		if app != nil {
+			logger.Info("Found Subscription-based application for DRPC",
+				"application", app.Name,
+				"subscriptionCount", len(subscriptions))
+			appRef = corev1.ObjectReference{
+				APIVersion: "app.k8s.io/v1beta1",
+				Kind:       "Application",
+				Name:       app.Name,
+				Namespace:  app.Namespace,
+				UID:        app.UID,
+			}
+		} else {
+			logger.Info("Subscriptions found but no parent Application",
+				"subscriptionCount", len(subscriptions))
+		}
+
+		return multiclusterv1alpha1.SubscriptionType, appRef, subRefs, nil
+	}
+
+	// No ApplicationSet or Subscription found - use Discovered type
+	logger.Info("No ApplicationSet or Subscription found, using Discovered type for DRPC")
+	return multiclusterv1alpha1.DiscoveredType, corev1.ObjectReference{
+		APIVersion: ramenAPIVersion,
+		Kind:       drpcKind,
+		Name:       drpc.Name,
+		Namespace:  drpc.Namespace,
+	}, nil, nil
+}
+
+// findApplicationSetByPlacement finds the ApplicationSet associated with the given Placement
+func (r *ProtectedApplicationViewReconciler) findApplicationSetByPlacement(
+	ctx context.Context,
+	drpc *ramenv1alpha1.DRPlacementControl,
+	placement *placementv1beta1.Placement,
+) (*argov1alpha1.ApplicationSet, error) {
+	logger := r.Logger.With("Placement", placement.Name, "DRPC", drpc.Name)
+
+	appSetList := &argov1alpha1.ApplicationSetList{}
+	if err := r.List(ctx, appSetList,
+		client.InNamespace(placement.Namespace),
+		client.MatchingFields{clusterDecisionResourceIndexName: placement.Name},
+	); err != nil {
+		logger.Error("Failed to list ApplicationSets", "error", err)
+		return nil, err
+	}
+
+	if len(appSetList.Items) == 0 {
+		return nil, nil
+	}
+
+	for i := range appSetList.Items {
+		appSet := &appSetList.Items[i]
+		if strings.HasPrefix(drpc.Name, appSet.Name) {
+			logger.Info("Found ApplicationSet matching naming convention",
+				"applicationSet", appSet.Name)
+			return appSet, nil
+		}
+	}
+
+	// No match found - log all names for debugging
+	names := make([]string, len(appSetList.Items))
+	for i, item := range appSetList.Items {
+		names[i] = item.Name
+	}
+
+	logger.Warn("No ApplicationSet matches naming convention, returning first",
+		"drpcName", drpc.Name,
+		"selectedAppSet", appSetList.Items[0].Name,
+		"allAppSets", names)
+
+	return &appSetList.Items[0], nil
+}
+
+func (r *ProtectedApplicationViewReconciler) findPAVsForDRPC(
+	ctx context.Context,
+	obj client.Object,
+) []reconcile.Request {
+	drpc, ok := obj.(*ramenv1alpha1.DRPlacementControl)
+	if !ok {
+		return nil
+	}
+
+	// PAV has same name/namespace as DRPC
+	return []reconcile.Request{{
+		NamespacedName: types.NamespacedName{
+			Name:      drpc.Name,
+			Namespace: drpc.Namespace,
+		},
+	}}
```

Comment:
This is assuming that the PAV name is always the same as the drpc name; Having assumption based on names always causes issues in future. We can list all the PAV and find out the PAV that is owned by the DRPC.

Reply:
I want to stick with the simpler approach for now. DRPC controller is creating the PAV so we can be assured that it will be named the same.